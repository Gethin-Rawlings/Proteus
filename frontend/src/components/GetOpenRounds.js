import React, { Component } from 'react';


const urlForNetworks = network => 'http://ec2-52-56-248-133.eu-west-2.compute.amazonaws.com:5000/organisations?type=open'

class GetOpenRounds extends Component {
  constructor(props) {
    super(props)
    this.state = {
      requestFailed: false,
      value: ''
    }
    this.handleChange = this.handleChange.bind(this);
  }
  handleChange(event) {
    const target = event.target;
    const name = target.name;
    const value = target.value;
    this.setState({ [name]: value });
  }
  componentDidMount() {
    fetch(urlForNetworks(this.props.network))
      .then(response => {
        if (!response.ok) {
          throw Error("Network request failed")
        }
        return response
      })
      .then(d => d.json())

      .then(d => {
        this.setState({
          Getopenrounds: JSON.stringify(d)
        })
      }, () => {
        this.setState({
          requestFailed: true
        })
      })
  }
  render() {
    if (this.state.requestFailed) return <p>Failed!</p>
    if (!this.state.Getopenrounds) return <p>Loading...</p>
    let returnData = JSON.parse(this.state.Getopenrounds)
    if (returnData.length === 0) return <select id="Openrounds"><option>No open rounds</option></select>
    return (
      <select id="Openrounds">
        {returnData.map(p => <option name="Openrounds" value={p.org_organisation_id} onChange={this.handleChange}>{p.org_description}</option>)}
      </select>
    )
  }
}

export default GetOpenRounds;
