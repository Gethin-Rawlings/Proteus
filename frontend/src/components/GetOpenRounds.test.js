import React from 'react';
import {shallow, mount} from 'enzyme';


import GetOpenRounds from './GetOpenRounds'

jest.mock("./apiCalls")
 test('GetOpenRounds renders as expected' ,() => {
    const component = shallow(<GetOpenRounds />);
    const tree = component;
    expect(tree).toMatchSnapshot();
 })


 jest.mock("./apiCalls")
    it('should respond to change event and change the state of the networks drop down', done => {        
      const wrapper = mount(<GetOpenRounds />);
      const data =  jest.fn()
      setTimeout(()=>{
      wrapper.update();
      console.log(wrapper.debug());
      wrapper.find('.network').simulate('change', { target: { Getnetworks: 'data', value: 'blah@gmail.com' } });
      expect(wrapper.state('data')).toEqual('blah@gmail.com');
    } , done())
})