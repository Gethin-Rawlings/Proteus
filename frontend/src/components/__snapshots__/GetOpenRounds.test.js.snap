// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetOpenRounds renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GetOpenRounds />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Failed!",
    },
    "ref": null,
    "rendered": "Failed!",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Failed!",
      },
      "ref": null,
      "rendered": "Failed!",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
